atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/BatchSizeRecorder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.autobatch;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.codahale.metrics.Histogram;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+
+@NotThreadSafe // Disruptor runs the batching function on just one thread.
+public final class BatchSizeRecorder {
+    static final String AUTOBATCHER_METER = "autobatcherMeter";
+
+    private final Histogram histo;
+
+    private BatchSizeRecorder(Histogram histo) {
+        this.histo = histo;
+    }
+
+    public static BatchSizeRecorder create(String safeLoggerIdentifier) {
+        Histogram meter = SharedTaggedMetricRegistries.getSingleton().histogram(MetricName.builder()
+                        .safeName(AUTOBATCHER_METER)
+                        .putSafeTags("safeIdentifier", safeLoggerIdentifier)
```

Comment:
nit: `safeTags` are required to be safe by definition, maybe `identifier`, unless the `safeIdentifier` key is used elsewhere

Reply:
Yep, it makes sense for the arg name. We probably named it this way because all the internal parameters for this field were named `safeIdentifier` to make it obvious.

Reply:
This was from the old version of the code, I tried to make as few possibly breaking changes as possible. Happy to call this identifier if we want to though, just wanted to explain the rationale.